.github/workflows/Ci.yml

```diff
@@ -0,0 +1,102 @@
+        ROS_2_ENV: "/opt/ros/foxy/setup.bash"
+        TERM: xterm
+      options: "--user root"
+
+    steps:
+    - name: Checkout ${{ github.event.repository.name }}
+      # Check out the repository code
+      uses: actions/checkout@v3.3.0
+      with:
+        path: src/${{ github.event.repository.name }}
+        fetch-depth: 0
+
+    - name: Move source code
+      # Move the source code to the desired location
+      run: mv $GITHUB_WORKSPACE/src /opt/carma/
+
+    - name: Checkout dependencies
+      # Checkout project dependencies
+      run: |
+        source "$INIT_ENV"
+        ./src/${{ github.event.repository.name }}/docker/checkout.bash -r /opt/carma/
+
+    - name: Build ROS1
+      # Build ROS1 packages
+      run: |
+        source "$INIT_ENV"
+        PACKAGES=$(find . -maxdepth 2 -type f -name package.xml | sed 's/\.\///' | cut -d/ -f1)
+        sed -i 's/colcon build /colcon build --packages-skip novatel_oem7_msgs tracetools tracetools_test /' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash
+        make_with_coverage.bash -m -e /opt/carma/ -o ./coverage_reports/gcov
+
+    - name: Run ROS1 C++ Tests
+      # Run ROS1 C++ tests
+      run: |
+        source "$INIT_ENV"
+        sed -i 's/colcon test /colcon test --packages-skip novatel_oem7_msgs tracetools tracetools_test /' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash
+        make_with_coverage.bash -t -e /opt/carma/ -o ./coverage_reports/gcov 
+
+    - name: Backup ROS1 compile_commands.json
+      # Backup the ROS1 compile_commands.json file for sonarcloud
+      run: |
+        mv /opt/carma/build/compile_commands.json /opt/carma/compile_commands.ros1.json
+
+    - name: Cleanup before ROS 2 build
+      # Clear the build and install folders before building ROS 2 
+      run: |
+        rm -rf /opt/carma/install
+        rm -rf /opt/carma/build             
+
+    - name: Build ROS2
+      # Build ROS2 packages
+      run: |
+        source "$INIT_ENV"
+        source "$ROS_2_ENV"
+        sed -i '/colcon build/ s/$/ --parallel-workers 4 --event-handlers console_direct+ --cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_CXX_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_C_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_BUILD_TYPE="Debug"/' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash
+        make_with_coverage.bash -m -e /opt/carma/ -o ./coverage_reports/gcov
+
+    - name: Run Ros2 C++ Tests
+      # Run ROS2 C++ tests
+      continue-on-error: true
+      run: |
+        source "$INIT_ENV"
+        source "$ROS_2_ENV"
+        sed -i '/colcon test/ s/$/ --packages-skip novatel_oem7_msgs approximate_intersection --parallel-workers 4 --event-handlers console_direct+ --cmake-args -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_CXX_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_C_FLAGS="${COVERAGE_FLAGS}" -DCMAKE_BUILD_TYPE="Debug"/' /home/carma/.ci-image/engineering_tools/code_coverage/make_with_coverage.bash

```

Comment:
Is there a reason we are skipping tests for `novatel_oem7_msgs` and `approximate_intersection` here?

Reply:
Removed the Package skip argument with packages